templates/common/fs_test.go

```diff
@@ -114,6 +120,19 @@ func TestCopyRecursive(t *testing.T) {
 			mkdirAllErr: fmt.Errorf("MkdirAll shouldn't be called in dry run mode"),
 			wantErr:     "file file1.txt already exists and overwriting was not enabled",
 		},
+		{
+			name: "dry_run_should_calculate_hashes",
+			srcDirContents: map[string]ModeAndContents{
+				"file1.txt": {Mode: 0o600, Contents: "file1 contents"},
+			},
+			hash:   sha256.New(),
+			dryRun: true,
+			wantHashes: map[string][]byte{
+				"file1.txt": mustHexDecode(t, "226e7cfa701fb8ba542d42e0f8bd3090cbbcc9f54d834f361c0ab8c3f4846b72"),
```

Comment:
Nit: it might be cleaner if `wantHashes` is `map[string]string` and you perform the hex decoding in the test function before comparison.

Reply:
I was on the fence about that too... given your vote, I'll go with hex. Done.

Reply:
I ended up just returning the hashes as hex.